### PR TITLE
Convert `/src/components/views/rooms` to TS

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: task

<hr>

This leaves out group components for obvious reasons 

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://61497bb145700e75f0e4feeb--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/E2EIcon.tsx [hunk 1]
```diff
@@ -16,33 +16,43 @@ limitations under the License.
...
-    NORMAL: "normal",
-    UNAUTHENTICATED: "unauthenticated",
-};
+export enum E2E_STATE {
+    VERIFIED = "verified",
+    WARNING = "warning",
+    UNKNOWN = "unknown",
+    NORMAL = "normal",
+    UNAUTHENTICATED = "unauthenticated",
+}
```
Comment (turt2live) [inline src/components/views/rooms/E2EIcon.tsx:32]: Should be updated to new naming conventions `E2EState.Verified` etc

**Review by turt2live — CHANGES_REQUESTED** [review]
otherwise looks good - thanks! Another one for post-RC :)

**Review by turt2live — COMMENTED** [review]
The `E2E_STATE` enum should also be renamed to `E2EState` to follow the convention used in practice

**Comment by SimonBrandner** [comment]
> The `E2E_STATE` enum should also be renamed to `E2EState` to follow the convention used in practice

Oops :man_facepalming:

**Review by turt2live — APPROVED** [review]

**Pull request merged**